android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -91,87 +95,123 @@ public Plugin() {
     public void load() {}
 
     /**
-     * Registers the permission launchers used by the {@link #requestPermissions(PluginCall)} plugin call and
-     * those defined on plugins
+     * Registers activity result launchers defined on plugins, used for permission requests and
+     * activities started for result.
      */
-    void initializePermissionLaunchers() {
-        basePermissionLauncher =
-            bridge
-                .getActivity()
-                .registerForActivityResult(
-                    new ActivityResultContracts.RequestMultiplePermissions(),
-                    permissions -> {
-                        PluginCall savedPermissionCall = bridge.getPermissionCall(handle.getId());
-
-                        if (bridge.validatePermissions(this, savedPermissionCall, permissions)) {
-                            checkPermissions(savedPermissionCall);
-
-                            if (!savedPermissionCall.isReleased() && !savedPermissionCall.isSaved()) {
-                                savedPermissionCall.release(bridge);
-                            }
-                        }
-                    }
-                );
+    void initializeActivityLaunchers() {
+        try {
+            // load the default checkPermission callback from the plugin parent class
+            Method method = getClass().getSuperclass().getMethod("checkPermissions", PluginCall.class);
```

Comment:
Did we decide not to get this dynamically via the annotations in the `Plugin` superclass?

Reply:
Yes, thank you. Fixed.